src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -81,4 +89,39 @@ public Date deserialize(JsonElement jsonElement, Type type,
             }
         }
     }
+
+    /**
+     * Deserializer for the Stats object received from the API.
+     */
+    private class StatsDeserializer implements JsonDeserializer<Stats> {
+
+        private static final String CP_APPLICATIONS = "CherryPy Applications";
+        private static final String CP_SERVER_PREFIX = "CherryPy HTTPServer ";
+
+        @Override
+        public Stats deserialize(JsonElement jsonElement, Type type,
+                JsonDeserializationContext jsonDeserializationContext)
+                throws JsonParseException {
+            try {
+                JsonObject stats = jsonElement.getAsJsonObject();
+                Applications app = gson.fromJson(stats.get(CP_APPLICATIONS), Applications.class);
+                HttpServer server = null;
+                for(Map.Entry<String, JsonElement> entry : stats.entrySet()){
+                    String key = entry.getKey();
+                    if(key.startsWith(CP_SERVER_PREFIX)){
+                        server = gson.fromJson(entry.getValue(), HttpServer.class);
+                        break;
+                    }
+                }
+                if(server != null) {
+                    return new Stats(app, server);
+                } else {
+                    throw new JsonParseException("CherryPy HTTPServer is null");
```

Comment:
I think `HTTP server is not available` gives the wrong impression because obviously it just returned me some `json` so the  HttpServer must be up and running. I don't think a valid response would ever have the server missing.